### PR TITLE
Do not invoke completion items list on typing inside unbounded argument

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

**Review by valfirst — APPROVED** [review]

File: vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java [hunk 1]
```diff
@@ -122,4 +134,57 @@ void testDidSave()
...
+        "new text, 1"
+    })
+    void shouldFilterCompletionItemsBasedOnInvocationType(String newText, int expectedSize)
+            throws InterruptedException, ExecutionException
+    {
+        CompletionParams params = mock(CompletionParams.class);
+        CompletionContext context = mock(CompletionContext.class);
+        CompletionItem item = mockCompletionItem(newText);
+        TextDocumentIdentifier identifier = mock(TextDocumentIdentifier.class);
+        Position position = spy(Position.class);
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java:150]:
```suggestion
        var params = mock(CompletionParams.class);
        var context = mock(CompletionContext.class);
        var item = mockCompletionItem(newText);
        var identifier = mock(TextDocumentIdentifier.class);
        var position = spy(Position.class);
```

File: vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java [hunk 2]
```diff
@@ -122,4 +134,57 @@ void testDidSave()
...
+        when(position.getCharacter()).thenReturn(11);
+
+        when(params.getContext()).thenReturn(context);
+        when(context.getTriggerKind()).thenReturn(CompletionTriggerKind.Invoked);
+        when(params.getTextDocument()).thenReturn(identifier);
+        when(identifier.getUri()).thenReturn(TEXT_DOCUMENT_IDENTIFIER);
+        when(params.getPosition()).thenReturn(position);
+        when(completionItemService.findAllAtPosition(TEXT_DOCUMENT_IDENTIFIER, position)).thenReturn(List.of(item));
+
+        DidChangeTextDocumentParams docParams = mockDidChange("O", 10);
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java:160]:
```suggestion
        var docParams = mockDidChange("O", 10);
```

File: vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java [hunk 3]
```diff
@@ -122,4 +134,57 @@ void testDidSave()
...
+
+        List<CompletionItem> items = textDocumentService.completion(params).get().getLeft();
+
+        assertThat(items, hasSize(expectedSize));
+    }
+
+    private static DidChangeTextDocumentParams mockDidChange(String text, int character)
+    {
+        DidChangeTextDocumentParams docParams = mock(DidChangeTextDocumentParams.class);
+        TextDocumentContentChangeEvent changeEvent = mock(TextDocumentContentChangeEvent.class);
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java:171]:
```suggestion
        var docParams = mock(DidChangeTextDocumentParams.class);
        var changeEvent = mock(TextDocumentContentChangeEvent.class);
```

Comment (ikalinin1) [reply vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/service/VividusStudioTextDocumentServiceTests.java:171]: etc

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**